docs/design/FreeCreditOfferTypesDesignDoc.md

```diff
@@ -0,0 +1,229 @@
+
+### CATEGORIES OF FREE CREDIT & OFFER TYPES
+
+**Default Free Tier Offer:**
+  - This is the credit that all new users who join the network receive if they create an account without a Referral Link. Once a new user successfully creates their account they will have 14 days to use their Free Credit allowance.  
+  The Default Offer will remain constant, with no expiration date to the offer and does not have a redemption cap
+  The Offer Program Admin is able to adjust the default offer
+  One Default Offer per Tardigrade Satellite.
+  When there is a lapse of the Referral Testing Offer/the Referral Testing Offer has expired or reached a cap, the Default Offer will be the default to the Referral Testing Offer that new users will redeem until a new Referral Testing Offer is generated.
+
+**Referral Testing Offer**:
+  - This is the credit awarded to the Invitee upon creating their account using a Referral Link. This credit can have a different expiration date than the awarded Referrer Credit and the Default Free Credit.
+  An admin of the Referral Program is able to easily adjust the current offer through the  Admin Satellite GUI
+
+**Referrer Award Credit**:
+  - This is the credit awarded to the Referrer once the Invitee has created a *paid* account and has successfully paid their second invoice. This credit can have a different expiration date than the awarded Invitee Credit and the Default Free Credit.
+
+[Offer Types Table](https://docs.google.com/spreadsheets/d/1I3Do-HMNkpUpJAsebtl1NXw6-PlkoVaFKCP9e-TTgHA/edit?zx=aqgnyh2ltfad#gid=0&range=A3:E13)
+
+## Design
+
+### Database
+
+**offer  table**
+```sql
+    Id - bytea
+    Name -  text
+    Description - text  
+    Credits - integer
+    redeemable_cap - integer
+    Num_redeemed - integer
+    Created_at - timestamp
+    offer_expiration - int
+    award_credit_expiration - int
+    Invitee_credit_expiration - int
+    type - enum[FREE_TIER, REFERRAL]
+    status - enum[ON_GOING, DEFAULT, EXPIRED, NO_STATUS]
+    PRIMARY KEY (id)
+```
+
+**user_credit_stats table**
+```sql
+    User_id - bytea
+    offer_id - bytea
+    credits_earned - float
+    Credit_type - enum[AWARD, INVITEE, NO_TYPE]
+    Is_expired - bool
+    Created_at - timestamp
+    PRIMARY KEY (user_id)
+```
+
+**user table**
+```sql
+    total_Referred - int
+```
+
+### Offer Program Service
+
+**satellite/offer/offer.go**
+- Create offers interface to interact with offer table
+
+```golang
+type Offers interface {
+  GetAllOffers()
+  GetOfferById(offerId)
+  Update(offerId)
+  Delete(offerId)
+  Create()
+}
+```
+
+**satellite/offer/credit.go**
+- Create a user_credit_stats interface to interact with the user_credit_stats  table
+
+```golang
+type user_credit_stats interface {
+  GetAvailableCreditsByUserId(userId)
+  GetUserCreditsByCreditType(userId, creditType)
+  Update(userId, offerId, creditType, isExpired)
+  Create(credit *UserCredit)
+}
+```
+
+**satellite/console/userCreditStats.go**
```

Comment:
nitpick, but the codebase usually uses all underscore snake case for file names.